File: mc/rcmd/src/main/java/calebxzhou/rdi/mc/rcmd/RcmdSource.java
package calebxzhou.rdi.mc.rcmd;

import java.util.UUID;

public interface RcmdSource {
    UUID NO_PLAYER_ID = new UUID(0L, 0L);

    String name();

    UUID playerId();

    default boolean isPlayer() {
        return !NO_PLAYER_ID.equals(playerId());
    }

    boolean hasPermission(String permission);

    void sendFeedback(String message);

    void sendError(String message);
}

File: mc/rcmd/src/main/kotlin/calebxzhou/rdi/mc/rcmd/RcmdSource.kt
package calebxzhou.rdi.mc.rcmd

import java.util.UUID

interface RcmdSource {
    fun name(): String

    fun playerId(): UUID

    val isPlayer: Boolean
        get() = NO_PLAYER_ID != playerId()

    fun hasPermission(permission: String): Boolean

    fun sendFeedback(message: String)

    fun sendError(message: String)

    companion object {
        @JvmField
        val NO_PLAYER_ID: UUID = UUID(0L, 0L)
    }
}
